src/CommunityToolkit.Mvvm.SourceGenerators/Diagnostics/DiagnosticDescriptors.cs

```diff
@@ -637,4 +642,20 @@ internal static class DiagnosticDescriptors
         isEnabledByDefault: true,
         description: "All attributes targeting the generated field or property for a method annotated with [RelayCommand] must be using valid expressions.",
         helpLinkUri: "https://aka.ms/mvvmtoolkit/errors/mvvmtk0038");
+
+    /// <summary>
+    /// Gets a <see cref="DiagnosticDescriptor"/> indicating when a method with <c>[RelayCommand]</c> is async void.
+    /// <para>
+    /// Format: <c>"The method {0} annotated with [RelayCommand] is async void (make sure to return a Task type instead)"</c>.
+    /// </para>
+    /// </summary>
+    public static readonly DiagnosticDescriptor AsyncVoidReturningRelayCommandMethod = new DiagnosticDescriptor(
+        id: AsyncVoidReturningRelayCommandMethodId,
+        title: "Async void returning method annotated with RelayCommand",
+        messageFormat: "The method {0} annotated with [RelayCommand] is async void (make sure to return a Task type instead)",
+        category: typeof(RelayCommandGenerator).FullName,
```

Comment:
I see this is already existing for all other descriptors, but I never seen categories to be very long names (`CommunityToolkit.Mvvm.SourceGenerators.RelayCommandGenerator`)

Reply:
Mmh good point. We might want to address this in a separate PR for all descriptors 🤔

Reply:
Just a note: this will be a breaking change if someone is currently changing severity via `dotnet_analyzer_diagnostic.category-TheLongCategoryName.severity = ...`